lib/beaker-hostgenerator/generator.rb

```diff
@@ -104,6 +112,19 @@ def __parse_node_info_token(token)
         node_info['arbitrary_roles'] = []
       end
 
+      if node_info['host_settings']
+        # Turns a string like "{foo=bar,this=that}" into a
+        # hash like {'foo' => 'bar', 'this' => 'that'}.
+        node_info['host_settings'] = Hash[
+          node_info['host_settings'].
+          delete('{}').
+          split(',').
+          map { |keyvalue| keyvalue.split('=') }
+        ]
+      else
+        node_info['host_settings'] = {}
+      end
+
       return node_info
     end
```

Comment:
I made several attempts to decompose this method and the larger `generate` method above, and I didn't like any of them.

I did change this code though, by moving it along with other tightly-coupled functions into the new Parser class.
I don't think decomposing `parse_node_info_token` any further would help much as it's not a deep method at all, just some datatype adjustments on a map.